packages/react-next/src/Utilities/Customizer.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import {
+  Customizations,
+  CustomizerContext,
+  mergeCustomizations,
+  ICustomizerContext,
+} from 'office-ui-fabric-react/lib/Utilities';
+import { ThemeProvider } from '@fluentui/react-theme-provider';
+import { convertLegacyTheme } from '../Styling/convertLegacyTheme';
+
+import { ICustomizerProps as LegacyCustomizerProps } from 'office-ui-fabric-react/lib/Utilities';
+
+export interface ICustomizerProps extends LegacyCustomizerProps {
+  disableThemeProvider?: boolean;
+}
+
+/**
+ * Replacement for Customizer from utilities package. It also provides the theme context.
+ *
+ * TODO (xgao): deprecate this before shipping react-next.
+ * Customizer should be replaceable by ThemeProvider and compose.
+ */
+export const Customizer: React.FunctionComponent<ICustomizerProps> = props => {
+  const forceUpdate = useForceUpdate();
+
+  React.useEffect(() => {
+    Customizations.observe(forceUpdate);
+
+    return () => Customizations.unobserve(forceUpdate);
+  }, []);
+
+  const { contextTransform, children, disableThemeProvider } = props;
+  return (
+    <CustomizerContext.Consumer>
+      {(parentContext: ICustomizerContext) => {
+        let newContext = mergeCustomizations(props, parentContext);
+
+        if (contextTransform) {
+          newContext = contextTransform(newContext);
+        }
+
+        const content = disableThemeProvider ? (
+          children
+        ) : (
+          <ThemeProvider theme={convertLegacyTheme(newContext.customizations.settings.theme)}>
```

Comment:
This seems backwards; shouldn't ThemeProvider (in react-next) include Customizer?

I might be wrong.

Reply:
The reason I have a new `Customizer` is bc it's specifically added for keeping backward compatibility with current `Customizer`. You can think of it as a shimmer and I actually want to ship this as `@deprecated`. I don't want to tight this with `ThemeProvider` and pollute its prop interface.

should `ThemeProvider` contain `Customizer`? I think the answer is "no" because we can use `ThemeContext` instead of `CustomizerContext`